test/Fencer/Server/Test.hs

```diff
@@ -72,20 +74,30 @@ test_responseNoRules =
 -- | Start Fencer on port 50051.
 createServer :: IO (Logger.Logger, ThreadId)
 createServer = do
+  (logger, threadId, _) <- createServerAppState
+  pure (logger, threadId)
+
+-- | Start Fencer on port 50051.
+createServerAppState :: IO (Logger.Logger, ThreadId, AppState)
+createServerAppState = do
   -- TODO: not the best approach. Ideally we should use e.g.
   -- https://hackage.haskell.org/package/tasty-hunit/docs/Test-Tasty-HUnit.html#v:testCaseSteps
   -- but we can't convince @tinylog@ to use the provided step function.
   logger <- Logger.create (Logger.Path "/dev/null")
   appState <- initAppState
   threadId <- forkIO $ runServer logger appState
-  pure (logger, threadId)
+  pure (logger, threadId, appState)
 
 -- | Kill Fencer.
 destroyServer :: (Logger.Logger, ThreadId) -> IO ()
 destroyServer (logger, threadId) = do
   Logger.close logger
   killThread threadId
 
+-- | Kill Fencer.
```

Comment:
I think the difference between `destroyServer` and `destroyServerAppState` should be detailed in the documentation. Or you might even get rid of `destroyServer`?

Reply:
I got rid of `destroyServer` yesterday and renamed `destroyServerAppState` to `destroyServre` because I realized that the usage of the old `destroyServer` does not care about the extra information that the `AppState`-version had.